tests: drivers: watchdog window reduced to fit stm32 board config

FRASTM opened the pull request:
Reduce the windows watchdog so that nucleo_f429 and nucleo_f746 can pass the tests/drivers/watchdog/wdt_basic_api whatever their sysclock and APB clock.

That change applies on the NUCLEO_F429ZI  target (cannot be a ./boards/nucleo_f429zi.overlay due to the existing stm32_wwdg.overlay)

Fixing the error when executing on the stm32f429zi or stm32f746zg nucleo boards
```
Running TESTSUITE wdt_basic_test_suite
===================================================================
START - test_wdt
Testcase: test_wdt_no_callback
Testcase passed
Testcase: test_wdt_callback_1
CB1 not supported on platform
Testcase: test_wdt_bad_window_max
PASS - test_wdt in 0.013 seconds
===================================================================
TESTSUITE wdt_basic_test_suite succeeded
------ TESTSUITE SUMMARY START ------
SUITE PASS - 100.00% [wdt_basic_test_suite]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.013 seconds
- PASS - [wdt_basic_test_suite.testRunning TESTSUITE wdt_basic_test_suite
===================================================================
START - test_wdt
Testcase: test_wdt_no_callback
Watchdog install error
Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/watchdog/wdt_basic_api/src/test_wdt.c:387: wdt_basic_test_suite_test_wdt: test_wdt_no_callback() == TC_PASS is false
FAIL - test_wdt in 0.021 seconds
===================================================================
TESTSUITE wdt_basic_test_suite failed.
------ TESTSUITE SUMMARY START ------
SUITE FAIL -   0.00% [wdt_basic_test_suite]: pass = 0, fail = 1, skip = 0, total = 1 duration = 0.021 seconds
- FAIL - [wdt_basic_test_suite.test_wdt] duration = 0.021 seconds
------ TESTSUITE SUMMARY END ------
===================================================================
RunID: e47f82fc05b160a9bf7204cfba4e659b
PROJECT EXECUTION FAILED

```

FRASTM committed:
tests: drivers: watchdog window reduced to fit stm32 board config

Reduce the windows watchdog so that any stm32 target
can pass the tests/drivers/watchdog/wdt_basic_api whatever
their sysclock and APB clock.
Especially nucleo_f429 and nucleo_f746

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on tests/drivers/watchdog/wdt_basic_api/src/test_wdt.c:
```diff
@@ -296,13 +297,13 @@ static int test_wdt_callback_1(void)
 	int err;
-	const struct device *const wdt = DEVICE_DT_GET(WDT_NODE);
+/*	const struct device *const wdt = DEVICE_DT_GET(WDT_NODE);
 
 	if (!device_is_ready(wdt)) {
 		TC_PRINT("WDT device is not ready\n");
 		return TC_FAIL;
 	}
-
+*/
```
Dev leftover ?

erwango commented on tests/drivers/watchdog/wdt_basic_api/src/test_wdt.c:
```diff
@@ -361,13 +362,13 @@ static int test_wdt_callback_2(void)
 {
 	int err;
-	const struct device *const wdt = DEVICE_DT_GET(WDT_NODE);
+/*	const struct device *const wdt = DEVICE_DT_GET(WDT_NODE);
 
 	if (!device_is_ready(wdt)) {
 		TC_PRINT("WDT device is not ready\n");
 		return TC_FAIL;
 	}
-
+*/
```
Dev leftover ?

erwango commented on tests/drivers/watchdog/wdt_basic_api/src/test_wdt.c:
```diff
@@ -71,7 +71,12 @@
 #elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32_window_watchdog)
 #define WDT_NODE DT_INST(0, st_stm32_window_watchdog)
 #define TIMEOUTS 0
+#if defined(CONFIG_BOARD_NUCLEO_F429ZI) || defined(CONFIG_BOARD_NUCLEO_F746ZG)
+/* Boards where the sysclock is high and APB1 prescaler 16 requires a lower WDG window */
+#define WDT_TEST_MAX_WINDOW 100
```
Anyway to make it less board specific ?

FRASTM commented:
Minimum and maximum timeout values depends on the watchdog peripheral : 
WWDG  prescaler (WDGTB)   1,2,4,8   for the stm32f42x serie
WWDG Timeout bit field  T[5:0] from 0 to 63 for the stm32f42x serie
and the boards frequency configuration : APB1 peripheral clock which depends on the  apb1 prescaler  = 1, 2 ,4, 8 or  16 stm32f42x serie
The max reachable value is  given by the formula
`t WWDG = t PCLK1 × 4096 × 2 ^ WDGTB[1:0] * ( T[5:0] + 1 )   ( in ms )`

For the nucleo_f429 where the sysclock is 168MHz and a max possible APB1 prescaler of 16:
t WWDG = 1/ (168  / 16 * 1000) * 4096 * 2³ * (63 +1) = 199 ms

Even with the apb1 prescaler set to 16, the nucleo_f429 (with a sysclock at 168MHz) cannot achieve WWDG timeout higher than 199.

Another possible configuration is to define the WDT_TEST_MAX_WINDOW at 190 ms  for testing the WWDG on all the stm32 targets

FRASTM committed:
boards: st: stm32h753 nucleo has a window watchdog

Enable the wwdg on the nucleo_h753zi board like it is
on the nucleo_h743zi

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
samples: drivers: watchdog window running on the stm32h7 nucleo

Add the nucleo_h753zi board when running the
samples/drivers/watchdog.
The wwdg is on APB3 clock tree (D1PPRE prescaler register)
(no need to set the d2ppre1)

Signed-off-by: Francois Ramu <francois.ramu@st.com>
